docs/CheckForApplicationUpdate.md

```diff
@@ -0,0 +1,142 @@
+---
+layout: default
+title: "The CheckForApplicationUpdate component"
+---
+
+# `CheckForApplicationUpdate`
+
+This component regularly checks whether the application source code has changed and prompt users to reload the page when an update is available. To detect updates, it fetches the current URL at regular intervals and compares a hash of the response content.
+
+It leverages the [`useCheckForApplicationUpdate`](./useCheckForApplicationUpdate.md) hook.
+
+![CheckForApplicationUpdate](./img/CheckForApplicationUpdate.png)
+
+## Usage
+
+Include this component in a custom layout:
+
+```tsx
+// in src/MyLayout.tsx
+import { CheckForApplicationUpdate, Layout, LayoutProps } from 'react-admin';
+
+export const MyLayout = ({ children, ...props }: LayoutProps) => (
+    <Layout {...props}>
+        <CheckForApplicationUpdate />
+    </Layout>
+);
+
+// in src/App.tsx
+import { Admin, ListGuesser, Resource } from 'react-admin';
+import { MyLayout } from './MyLayout';
+
+export const App = () => (
+    <Admin layout={MyLayout}>
+        <Resource name="posts" list={ListGuesser} />
+    </Admin>
+);
+```
+
+## Props
+
+`<CheckForApplicationUpdate>` accepts the following props:
+
+| Prop            | Required | Type     | Default            | Description                                                         |
+| --------------- | -------- | -------- | ------------------ |-------------------------------------------------------------------- |
+| `checkInterval` | Optional | number   | `3600000` (1 hour) | The interval in milliseconds between two checks                     |
+| `disabled`      | Optional | boolean  | `true`             | Whether the automatic check is enabled                              |
+| `notification`  | Optional | string   |                    | The notification to display to the user when an update is available |
```

Comment:
You say `notification` prop is a string but in the example below you pass a `ReactElement` 🤔